### PR TITLE
Apns cred check

### PR DESCRIPTION
**Pull request by viktormuller**
Checks that iOS app provisioning profile and credential is aligned.

This is the no.1 problem developer's run into hence an automated check should greatly help debugging. Needs corresponding changes in iOS apps as well.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,13 +24,42 @@ app.get('/', function(request, response) {
...
 
   // Get a reference to the user notification service instance
   var service = client.notify.v1.services(env.TWILIO_NOTIFICATION_SERVICE_SID);
 
+  // For APNS bindings let's check if the app's provisioning profile is aligned with that of the configured certificate.
+  //TODO: Please, refactor if necessary to reflect JS best practices
+  if (bindingType === "apn"){
+    var apnCredMatched = false;
+    service.fetch().then(function(serviceObj){
+      client.notify.v1.credentials(serviceObj.apnCredentialSid).fetch()
```
Comment (kwhinnery) [inline index.js]: return this promise, and move the `then` just before your final `catch` below - then there will be only one `catch` for this block.

**Comment by viktormuller** [comment]
@kwhinnery Thanks. I have tried to improve it. Also simplified the config by removing the apn and gcm credential SIDs as those are not used anymore. They are configured via the console and they can just cause confusion as they can be out of sync with actual config in console.

**Comment by viktormuller** [comment]
@coderkat This also supports now endpoint ID autogeneration by accepting registration requests without the Endpoint ID and including Endpoint ID in the response.

File: index.js [hunk 2]
```diff
@@ -14,47 +15,95 @@ app.use(express.static('public'));
 // Basic health check - check environment variables have been configured
 // correctly
 app.get('/', function(request, response) {
-  response.render('index.jade', {
+
```
Comment (coderkat-zz) [inline index.js:18]: Nitpick, but we don't need this blank line

File: index.js [hunk 3]
```diff
@@ -14,47 +15,95 @@ app.use(express.static('public'));
...
+  response.render('index.jade',{
     TWILIO_ACCOUNT_SID: env.TWILIO_ACCOUNT_SID,
     TWILIO_AUTH_TOKEN: env.TWILIO_AUTH_TOKEN,
     TWILIO_NOTIFICATION_SERVICE_SID: env.TWILIO_NOTIFICATION_SERVICE_SID,
+
```
Comment (coderkat-zz) [inline index.js:23]: Nitpick: blank line

File: index.js [hunk 4]
```diff
@@ -14,47 +15,95 @@ app.use(express.static('public'));
...
   });
+
```
Comment (coderkat-zz) [inline index.js:25]: Nitpick: blank line

**Review by coderkat-zz — COMMENTED** [review]
Looks good! Just a few very nitpicky style comments.

**Comment by viktormuller** [comment]
@coderkat Thx. Done. Sorry for the delay have not seen the comments.

**Pull request merged**